### ISSUE TITLE
The required version to use the AdMOB SDK is 16 or higher

### ISSUE THREAD

**Issue by nemoryoliver** [bdlukaa/native_admob_flutter#8]
Using the latest version and getting this error

`I/flutter ( 8746): ══╡ EXCEPTION CAUGHT BY WIDGETS LIBRARY ╞═══════════════════════════════════════════════════════════
I/flutter ( 8746): The following assertion was thrown building NativeAd(dirty, state: _NativeAdState#1861b):
I/flutter ( 8746): The required version to use the AdMOB SDK is 16 or higher
I/flutter ( 8746): 'package:native_admob_flutter/src/utils.dart':
I/flutter ( 8746): Failed assertion: line 47 pos 7: 'MobileAds.osVersion >= 16'
I/flutter ( 8746):
I/flutter ( 8746): The relevant error-causing widget was:
I/flutter ( 8746):   NativeAd
I/flutter ( 8746):
lib/…/main/main.screen.dart:226
I/flutter ( 8746):
I/flutter ( 8746): When the exception was thrown, this was the stack:
I/flutter ( 8746): #2      assertVersionIsSupported
package:native_admob_flutter/src/utils.dart:47
I/flutter ( 8746): #3      _NativeAdState.build
package:native_admob_flutter/…/native/native_ad_widget.dart:219
I/flutter ( 8746): #4      StatefulElement.build
package:flutter/…/widgets/framework.dart:4744
I/flutter ( 8746): #5      ComponentElement.performRebuild
package:flutter/…/widgets/framework.dart:4627
I/flutter ( 8746): #6      StatefulElement.performRebuild
package:flutter/…/widgets/framework.dart:4800
I/flutter ( 8746): #7      Element.rebuild
package:flutter/…/widgets/framework.dart:4343
I/flutter ( 8746): #8      ComponentElement._firstBuild
package:flutter/…/widgets/framework.dart:4606
I/flutter ( 8746): #9      StatefulElement._firstBuild
package:flutter/…/widgets/framework.dart:4791
I/flutter ( 8746): #10     ComponentElement.mount
package:flutter/…/widgets/framework.dart:4601
I/flutter ( 8746): ...     Normal element mounting (13 frames)
I/flutter ( 8746): #23     Element.inflateWidget
package:flutter/…/widgets/framework.dart:3569
I/flutter ( 8746): #24     MultiChildRenderObjectElement.mount
package:flutter/…/widgets/framework.dart:6236
I/flutter ( 8746): ...     Normal element mounting (91 frames)
I/flutter ( 8746): #115    Element.inflateWidget
package:flutter/…/widgets/framework.dart:3569
I/flutter ( 8746): #116    Element.updateChild
package:flutter/…/widgets/framework.dart:3324
I/flutter ( 8746): #117    ComponentElement.performRebuild
package:flutter/…/widgets/framework.dart:4652
I/flutter ( 8746): #118    StatefulElement.performRebuild
package:flutter/…/widgets/framework.dart:4800
I/flutter ( 8746): #119    Element.rebuild
package:flutter/…/widgets/framework.dart:4343
I/flutter ( 8746): #120    BuildOwner.buildScope
package:flutter/…/widgets/framework.dart:2730
I/flutter ( 8746): #121    WidgetsBinding.drawFrame
package:flutter/…/widgets/binding.dart:913
I/flutter ( 8746): #122    RendererBinding._handlePersistentFrameCallback
package:flutter/…/rendering/binding.dart:302
I/flutter ( 8746): #123    SchedulerBinding._invokeFrameCallback
package:flutter/…/scheduler/binding.dart:1117
I/flutter ( 8746): #124    SchedulerBinding.handleDrawFrame
package:flutter/…/scheduler/binding.dart:1055
I/flutter ( 8746): #125    SchedulerBinding._handleDrawFrame
package:flutter/…/scheduler/binding.dart:971
I/flutter ( 8746): #129    _invoke (dart:ui/hooks.dart:251:10)
I/flutter ( 8746): #130    _drawFrame (dart:ui/hooks.dart:209:3)
I/flutter ( 8746): (elided 5 frames from class _AssertionError and dart:async)
I/flutter ( 8746):
I/flutter ( 8746): ════════════════════════════════════════════════════════════════════════════════════════════════════`

I guess the logic here supposed to be the opposite?

```
/// Assert the current version is supported.
/// The min versions are:
///  - iOS: 9
///  - Android: 16 (19 for Native and Banner Ads)
void assertVersionIsSupported([bool usePlatformView = true]) {
  if (Platform.isAndroid) {
    /// The min required version for Android is 16
    assert(
      MobileAds.osVersion >= 16,
      'The required version to use the AdMOB SDK is 16 or higher',
    );

    /// The required version by flutter to use PlatformViews is
    ///   - Hybrid composition: 19
    ///   - Virtual display: 20
    if (usePlatformView)
      assert(
        MobileAds.osVersion >= 19,
        'Native and Banner Ads are not supported in versions before 19 because'
        ' flutter only support platform views on Android 19 or greater.',
      );
  } else {
    /// The min required version for iOS is 9
    assert(
      MobileAds.osVersion >= 9,
      'The required version to use the AdMOB SDK is 9 or higher',
    );
  }
}
```

**Comment by bdlukaa** [comment]
What's the android version you're running on?

**Comment by nemoryoliver** [comment]
> What's the android version you're running on?

testing it on Android 11 OnePlus 8 device, as well as Android 8.1.0 BlackBerry KEY2 device

**Comment by ycv005** [comment]
I am too facing the same issue @bdlukaa . I am running on android version 10. Please fix it as soon as possible.

**Comment by bdlukaa** [comment]
> Reading the codebase, I found that you didn't using any logic to get correct version. (may be I would be wrong)

I actually check the current version on `MobileAds.initialize()`. This issue may be due to incorrect initialization. You can initialize it like in the [example app](https://github.com/bdlukaa/native_admob_flutter/blob/master/example/lib/main.dart). 

> as well as Android 8.1.0 BlackBerry KEY2 device

I am testing in a Android 9 device and it works just fine. 

```dart
_version = await _pluginChannel.invokeMethod<int>('initialize');
assertVersionIsSupported(false);
```

> I guess the logic here supposed to be the opposite?

I don't think so

**Comment by bdlukaa** [comment]
I just published a new version that is supposed to fix this issue: [v0.9.4](https://pub.dev/packages/native_admob_flutter)

I'll be closing this issue. If the problem is not solved, please re-open this issue with more details.
